### ISSUE TITLE
Update buttons that are role-based

### ISSUE THREAD

**Issue by thaafei** [thaafei/DomainX#391]
Currently, the majority of the buttons are accessible to everyone, regardless of role. These should be updated to reflect the accessibility level. I.e. Edit metrics should only be seen by admin+. Edit metric values edit/update buttons only by admin+, and etc. 

See implementation of the create domain button as reference
